gossip/server.go

```diff
@@ -88,6 +88,10 @@ func (s *server) Gossip(argsI proto.Message) (proto.Message, error) {
 	if err != nil {
 		return nil, util.Errorf("local addr %s could not be converted to net.Addr: %s", args.LAddr, err)
 	}
+	// Verify that the client connection is valid and hasn't been closed.
+	if _, ok := s.lAddrMap[lAddr.String()]; !ok {
```

Comment:
I also have the impression that tracking clients based on their local address is frail. Why don't clients identify with their (distinguished) gossip server's listening address (via payload)? I think that would even obviate the need for some of the `NodeID`-based tracking.